### PR TITLE
Implement `SequenceClassificationRewardModel`

### PR DESCRIPTION
**Pull request by ryokan0123**
## What's new?

* Implement `SequenceClassificationRewardModel`
  * Compare two chat responses based on scalar reward value.
## Changes to the code
* Extract the logic to build model kwargs from `HuggingFaceLM` as `get_default_model_kwargs`
  * To share it between `HuggingFaceLM` and `SequenceClassificationRewardModel` 
* Added `tests/dummy_modules/llama-seq-classification-tiny` for unit test
  * This is a small dummy model of `LlamaForSequenceClassification`

### CODE REVIEW

File: flexeval/core/reward_model/sequence_classification.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    ) -> None:
+        tokenizer = tokenizer if tokenizer else model
+        tokenizer_kwargs = tokenizer_kwargs or {}
+        self.tokenizer = AutoTokenizer.from_pretrained(tokenizer, **tokenizer_kwargs)
+
+        model_kwargs = get_default_model_kwargs(model_kwargs)
+        self.model = AutoModelForSequenceClassification.from_pretrained(model, **model_kwargs)
+        self.model.eval()
+
+        self.tokenizer = AutoTokenizer.from_pretrained(model)
```
Comment (Ktakuya332C) [inline flexeval/core/reward_model/sequence_classification.py:33]: Duplicate to the line 27. Typo?

Comment (ryokan0123) [reply flexeval/core/reward_model/sequence_classification.py]: Good catch!
Thank you, fixed.

https://github.com/sbintuitions/flexeval/pull/101/commits/3be1c8e520b86f408714025b4fe2fafa7e6281d6

**Review by Ktakuya332C — APPROVED** [review]
